### PR TITLE
Fix selection on 3D viewport with half resolution

### LINKED ISSUE #13364: Mouse click selection is broken for 'half resolution' viewport option
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
d992eb1b25e3612293d6c72cd462a99cd5a0e4a3

**Issue description:**
It's impossible to select proper object when 'half resolution' rendering option is checked in viewport settings. Usually some other object get selected. Selection code probably is not aware of the fact that viewport is rendered with half of the normal resolution.

**Steps to reproduce:**

**Link to minimal example project:**

### PR DESCRIPTION
**Pull request by JFonS**
Fixes #13364.

### CODE REVIEW

File: editor/plugins/spatial_editor_plugin.cpp [hunk 1]
```diff
@@ -298,6 +298,7 @@ ObjectID SpatialEditorViewport::_select_ray(const Point2 &p_pos, bool p_append,
 
 	Vector3 ray = _get_ray(p_pos);
 	Vector3 pos = _get_ray_pos(p_pos);
+	Vector2 shrinked_pos = p_pos/viewport_container->get_stretch_shrink();
```
Comment (akien-mga) [inline editor/plugins/spatial_editor_plugin.cpp]: Style issue.

**Comment by Calinou** [comment]
Does this make #33940 obsolete?

**Comment by akien-mga** [comment]
> Does this make #33940 obsolete?

It doesn't seem to, I tested and this PR does fix #13364 but #33937 is still reproducible.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**